src/patches/dense_patch_computation.cpp

```diff
@@ -289,9 +301,26 @@ InstructionApplicationResult try_apply_instruction_direct_followup(
         if (!slice.has_patch(target_id))
             return {std::make_unique<std::runtime_error>(lstk::cat(instruction,"; Patch ", target_id, " not on lattice")), {}};
 
-        if (!merge_patches(slice, router, source_id, source_op, target_id, target_op))
-            return {std::make_unique<std::runtime_error>(lstk::cat(instruction,"; Couldn't find room to route")), {}};
-        return {nullptr, {}};
+        if (!local_instructions) 
+        {
+            if (!merge_patches(slice, router, source_id, source_op, target_id, target_op))
+                return {std::make_unique<std::runtime_error>(lstk::cat(instruction,"; Couldn't find room to route")), {}};
+            return {nullptr, {}};
+        }
+        else
+        {
+            // TODO: break up long-range ops into local ops
+            std::vector<LocalInstruction::LocalLSInstruction> local_instructions;
+            local_instructions.push_back({LocalInstruction::TwoPatchMeasure{slice.get_cell_by_id(source_id).value(), slice.get_cell_by_id(target_id).value()}});
+            InstructionApplicationResult r = try_apply_local_instruction(slice, local_instructions[0], layout, router);
```

Comment:
Why are we creating a vector here? is it because you want to pass more than one instruction later?

Reply:
There would be a loop over LocalInstructions if local compilation were implemented for long-range MultiPatchMeasurements, but for now it is just treated as a single TwoPatchMeasure (and will fail for cells that do not share a boundary).